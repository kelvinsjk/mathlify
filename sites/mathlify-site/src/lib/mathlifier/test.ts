import { mathlify } from './';

const str = mathlify`This is inline math: ${'x+2'}xyz${'k'}
and this is text.

This is display math: $${'x+2'}xyz
abcd
${'k'}

This is aligned:
$${'align'}
  x+2 &= ${'k'} \\\\
  y &= ${'z'}`;

console.log(str);
