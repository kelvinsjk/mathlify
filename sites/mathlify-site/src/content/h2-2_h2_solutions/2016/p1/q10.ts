import { Answer } from '$lib/classes/answer';
import { fnTerm, sum } from 'mathlify';

import { sqrtInverse } from '$content/h2-1_h2_learn/01_functions/02_inverse/03_formula/02_practice';
import { generateFn } from '$content/h2-1_h2_learn/01_functions/01_concepts/02_domain-and-range/02_practice-1';
import { EquationWorking } from 'mathlify/working';
import { mathlifierDj as mathlifier } from 'mathlifier';
import { simplifySurd } from 'mathlify/fns';
import { bisection, cubicRoot } from 'mathlify/numerical';
import { QED } from '$lib/utils/typesetting/utils';

export const answer = new Answer();

// 1 + sqrt(x)

const [a, b] = [0, 1];

const state: Parameters<typeof sqrtInverse>[0] = {
	fnType: 'sqrt',
	a,
	b,
	restriction: false,
	// dummies
	definition: false,
	c: 1,
	unknownConstants: false
};
const f = generateFn(state)[1];

// ai
{
	const { ans, soln } = sqrtInverse(state, f, { definition: false });
	answer.addSubPart(ans, soln);
}
// aii
{
	const working = new EquationWorking(fnTerm('f', f), 'x');
	EquationWorking.RegisterCustomSimplifier(simplifySurd);
	working.subIn({ f }).moveTerms(1).square().expand().moveTerms(1);
	working.square().expand().moveTerms(0, { swapSides: true }).divide('x');
	EquationWorking.DeregisterCustomSimplifier();
	const cubic = working.eqn.lhs;

	const soln1 = mathlifier`$${'gather*'} ff(x) = x \\\\ ${working}`;
	const x2 = cubicRoot(cubic);
	const x1 = bisection(cubic.fn, 0, 1);
	const x3 = bisection(cubic.fn, 2, 3);

	const soln2 = mathlifier`Solving with a GC,

$${{}} x = ${x1.toPrecision(3)}, \\quad ${x2} \\quad \\text{or} \\quad ${x3.toPrecision(3)}

From ${{}} {\\sqrt{\\sqrt{x}+1}+1=x,}
we observe that ${{}} {x>1.}
Hence

$${{}} x = ${x3.toPrecision(3)} \\; ${QED} \n\n`;
	const soln3 = mathlifier`From ${{}}{ff(x)=x},
we can apply ${'f^{-1}'}
on both sides

$${'align*'} ff(x) &= x
\\\\ f^{-1}ff(x) &= f^{-1}(x)
\\\\ f(x) &= f^{-1}(x)

Hence the value of ${'x'}
obtained satisfies the equation ${{}} {f(x)=f^{-1}(x)} \\; ${QED}`;
	const ans = mathlifier`${{}} x = ${x3.toPrecision(3)}.`;
	answer.addSubPart(ans, soln1 + soln2 + soln3);
}
answer.newPart();
// b
{
	const evenCase = sum(2, fnTerm('g', [[1, '/', 2], 'n']));
	const oddCase = sum(1, fnTerm('g', sum('n', -1)));
	const soln = mathlifier`$${'align*'} g(2) &= ${evenCase.subIn({ n: 2 })}
\\\\ &= 2 + ${oddCase.subIn({ n: 1 })}
\\\\ &= 3 + 1
\\\\ &= 4

$${'align*'} g(4) &= ${evenCase.subIn({ n: 4 })}
\\\\ &= 2 + 4
\\\\ &= 6 \\; ${QED}

$${'align*'} g(6) &= ${evenCase.subIn({ n: 6 })}
\\\\ &= 2 + ${oddCase.subIn({ n: 3 })}
\\\\ &= 3 + 4
\\\\ &= 7

$${'align*'} g(7) &= ${oddCase.subIn({ n: 7 })}
\\\\ &= 1 + 7
\\\\ &= 8 \\; ${QED}

$${'align*'} g(12) &= ${evenCase.subIn({ n: 12 })}
\\\\ &= 2 + ${evenCase.subIn({ n: 6 })}
\\\\ &= 2 + 7
\\\\ &= 9 \\; ${QED}
`;
	const ans = mathlifier`${{}} g(4) = 6.
\\
${{}} g(7) = 8.
\\
${{}} g(12) = 9.`;
	answer.addSubPart(ans, soln);

	// bii
	const soln2 = mathlifier`$${'align*'} g(8) &= ${evenCase.subIn({ n: 8 })}
\\\\ &= 2 + 6
\\\\ &= 8

Since ${{}} {g(7)=g(8)=8,}
${'g'}
is not a one-to-one function and ${'g'}
does not have an inverse ${QED}`;
	const ans2 = mathlifier`${'g'}
does not have an inverse as ${{}}{g(7)=g(8)=8}
so ${'g'}
is not a one-to-one function.`;
	answer.addSubPart(ans2, soln2);
}
