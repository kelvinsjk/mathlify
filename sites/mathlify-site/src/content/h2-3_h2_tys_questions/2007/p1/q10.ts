/* The question here is the copyright of 
- Singapore Examinations and Assessment Board
- Cambridge Assessment International Education
- Ministry of Education, Singapore
- Government of Singapore

We provide this question on Mathlify under fair use for archival purposes.
These are questions more than 10 years old and are difficult to obtain
commercially.

For the newer questions, please purchase a copy of the
"ten year series" from authorized publishers and distributors.
*/

import { Question } from '$lib/classes/question';
import { mathlifierDj as mathlifier } from 'mathlifier';
import { Polynomial } from 'mathlify';

export const question = new Question();

const poly = new Polynomial([3, -5, 2], { variable: 'r' });

question.addBody(mathlifier`A geometric series has a common ratio ${'r'},
and an arithmetic series has first term ${'a'}
and common difference ${'d'},
where ${'a'}
and ${'d'}
are non-zero.

The first three terms of the geometric series are
equal to the first, fourth and sixth terms respectively of the arithmetic series`);

// a
question.addPart(mathlifier`Show that ${{}} {${poly}.}`, { marks: 4 });
// b
question.addPart(
	mathlifier`Deduce that the geometric series is convergent
and find, in terms of ${'a'},
the sum to infinity.`,
	{ marks: 5 }
);
// c
question.addPart(
	mathlifier`The sum of the first ${'n'}
terms of the arithmetic series is denoted by ${'S'}.
Given that ${{}} {a>0,}
find the set of possible values of ${'n'}
for which ${'S'}
exceeds ${'4a'}.`,
	{ marks: 5 }
);
