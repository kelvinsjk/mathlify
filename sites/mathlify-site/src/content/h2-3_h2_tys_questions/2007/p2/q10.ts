/* The question here is the copyright of 
- Singapore Examinations and Assessment Board
- Cambridge Assessment International Education
- Ministry of Education, Singapore
- Government of Singapore

We provide this question on Mathlify under fair use for archival purposes.
These are questions more than 10 years old and are difficult to obtain
commercially.

For the newer questions, please purchase a copy of the
"ten year series" from authorized publishers and distributors.
*/

import { Question } from '$lib/classes/question';
import { mathlifierDj as mathlifier } from 'mathlifier';
import { quotient } from 'mathlify';

export const question = new Question();

const p = quotient(1, 8);

question.addBody(mathlifier`A player throws three darts at a target.
The probability that he is successful in hitting the target with
his first throw is ${p}.
For each of his second and third throws, the probability of success is

- twice the probability of success on the preceding throw if that throw was successful,
- the same as the probability of success on the preceding throw if that throw was unsuccessful.`);
// a
question.addPart(mathlifier`Construct a probability tree showing this information.`, { marks: 3 });
// b
question.addPart(mathlifier`Find`);
question.addSubPart(`the probability that all three throws are successful,`, { marks: 2 });
question.addSubPart(`the probability that at least two throws are successful,`, { marks: 2 });
question.addSubPart(
	`the probability that the third throw is successful
given that exactly two of the three throws are successful.`,
	{ marks: 4 }
);
