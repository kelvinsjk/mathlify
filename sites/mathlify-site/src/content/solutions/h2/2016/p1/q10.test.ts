import { expect, it } from 'vitest';
import { answer } from './q10.soln';
import { mathlify } from '$lib/mathlifier';

const ans = answer.answer;
const actual: typeof ans = {
	parts: [
		{
			parts: [
				{
					body: mathlify`${{}} f^{-1}(x) = \\left( x - 1 \\right)^2.
\\
${{}} D_{f^{-1}} = R_f = \\left[1, \\infty \\right).`,
				},
				{ body: mathlify`${{}} x=2.62.` },
			],
		},
		{
			parts: [
				{
					body: mathlify`${{}} g(4) = 6.
\\
${{}} g(7)=8.
\\
${{}} g(12)=9.`,
				},
				{
					body: mathlify`${{}} g
does not have an inverse as ${{}} {g(7)=g(8)=8}
so ${'g'}
is not a one-to-one function.`,
				},
			],
		},
	],
};

it('2016p1q10', () => {
	expect(answer).toMatchSnapshot();
	expect(ans).toEqual(actual);
});
