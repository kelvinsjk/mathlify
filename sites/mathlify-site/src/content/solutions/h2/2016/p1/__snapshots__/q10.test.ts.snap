// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`2016p1q10 1`] = `
Answer {
  "answer": {
    "parts": [
      {
        "parts": [
          {
            "body": "<p><math><mrow><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo></mrow><mrow><msup><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>1</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup><mi>.</mi></mrow></math>
<br>
<math><mrow><msub><mi>D</mi><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup></msub><mo>=</mo></mrow><mrow><msub><mi>R</mi><mi>f</mi></msub><mo>=</mo></mrow><mrow><mrow><mo fence="true" form="prefix">[</mo><mn>1</mn><mo separator="true">,</mo><mi>∞</mi><mo fence="true" form="postfix">)</mo></mrow><mi>.</mi></mrow></math></p>
",
          },
          {
            "body": "<p><math><mrow><mi>x</mi><mo>=</mo></mrow><mrow><mn>2.62</mn><mi>.</mi></mrow></math></p>
",
          },
        ],
      },
      {
        "parts": [
          {
            "body": "<p><math><mrow><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mn>4</mn><mo form="postfix" stretchy="false">)</mo><mo>=</mo></mrow><mrow><mn>6</mn><mi>.</mi></mrow></math>
<br>
<math><mrow><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mn>7</mn><mo form="postfix" stretchy="false">)</mo><mo>=</mo></mrow><mrow><mn>8</mn><mi>.</mi></mrow></math>
<br>
<math><mrow><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mn>12</mn><mo form="postfix" stretchy="false">)</mo><mo>=</mo></mrow><mrow><mn>9</mn><mi>.</mi></mrow></math></p>
",
          },
          {
            "body": "<p><math><mi>g</mi></math>
does not have an inverse as <math><mrow><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mn>7</mn><mo form="postfix" stretchy="false">)</mo><mo>=</mo><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mn>8</mn><mo form="postfix" stretchy="false">)</mo><mo>=</mo><mn>8</mn></mrow></math>
so <math><mi>g</mi></math>
is not a one-to-one function.</p>
",
          },
        ],
      },
    ],
  },
  "solution": {
    "parts": [
      {
        "parts": [
          {
            "body": "<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true"><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mtext>Let </mtext><mi>y</mi><mo>=</mo><msqrt><mi>x</mi></msqrt><mo>+</mo><mn>1</mn></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msqrt><mi>x</mi></msqrt><mo>=</mo><mi>y</mi><mo>−</mo><mn>1</mn></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>x</mi><mo>=</mo><msup><mrow><mo fence="true" form="prefix">(</mo><mi>y</mi><mo>−</mo><mn>1</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup></mrow></mtd></mtr></mtable></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mrow><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo><msup><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>1</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup><mspace width="0.2778em"></mspace><mi>■</mi></mrow></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><msub><mi>D</mi><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup></msub></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><msub><mi>R</mi><mi>f</mi></msub></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mrow><mo fence="true" form="prefix">[</mo><mn>1</mn><mo separator="true">,</mo><mi>∞</mi><mo fence="true" form="postfix">)</mo></mrow><mspace width="0.2778em"></mspace><mi>■</mi></mrow></mtd></mtr></mtable></math></p>
",
          },
          {
            "body": "<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true"><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>f</mi><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo><mi>x</mi></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>f</mi><mrow><mo fence="true" form="prefix">(</mo><msqrt><mi>x</mi></msqrt><mo>+</mo><mn>1</mn><mo fence="true" form="postfix">)</mo></mrow><mo>=</mo><mi>x</mi></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msqrt><mrow><msqrt><mi>x</mi></msqrt><mo>+</mo><mn>1</mn></mrow></msqrt><mo>+</mo><mn>1</mn><mo>=</mo><mi>x</mi></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msqrt><mrow><msqrt><mi>x</mi></msqrt><mo>+</mo><mn>1</mn></mrow></msqrt><mo>=</mo><mi>x</mi><mo>−</mo><mn>1</mn></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msqrt><mi>x</mi></msqrt><mo>+</mo><mn>1</mn><mo>=</mo><msup><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>1</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msqrt><mi>x</mi></msqrt><mo>+</mo><mn>1</mn><mo>=</mo><msup><mi>x</mi><mn>2</mn></msup><mo>−</mo><mn>2</mn><mi>x</mi><mo>+</mo><mn>1</mn></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msqrt><mi>x</mi></msqrt><mo>=</mo><msup><mi>x</mi><mn>2</mn></msup><mo>−</mo><mn>2</mn><mi>x</mi></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>x</mi><mo>=</mo><msup><mrow><mo fence="true" form="prefix">(</mo><msup><mi>x</mi><mn>2</mn></msup><mo>−</mo><mn>2</mn><mi>x</mi><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>x</mi><mo>=</mo><msup><mi>x</mi><mn>4</mn></msup><mo>−</mo><mn>4</mn><msup><mi>x</mi><mn>3</mn></msup><mo>+</mo><mn>4</mn><msup><mi>x</mi><mn>2</mn></msup></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msup><mi>x</mi><mn>4</mn></msup><mo>−</mo><mn>4</mn><msup><mi>x</mi><mn>3</mn></msup><mo>+</mo><mn>4</mn><msup><mi>x</mi><mn>2</mn></msup><mo>−</mo><mi>x</mi><mo>=</mo><mn>0</mn></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msup><mi>x</mi><mn>3</mn></msup><mo>−</mo><mn>4</mn><msup><mi>x</mi><mn>2</mn></msup><mo>+</mo><mn>4</mn><mi>x</mi><mo>−</mo><mn>1</mn><mo>=</mo><mn>0</mn><mspace width="0.2778em"></mspace><mi>■</mi></mrow></mtd></mtr></mtable></math></p>
<p>Solving with a GC,</p>
<p><math display="block" class="tml-display" style="display:block math;"><mrow><mi>x</mi><mo>=</mo><mn>0.382</mn><mo separator="true">,</mo><mspace width="1em"></mspace><mn>1</mn><mspace width="1em"></mspace><mtext>or</mtext><mspace width="1em"></mspace><mn>2.62</mn></mrow></math></p>
<p>From <math><mrow><msqrt><mrow><msqrt><mi>x</mi></msqrt><mo>+</mo><mn>1</mn></mrow></msqrt><mo>+</mo><mn>1</mn><mo>=</mo><mi>x</mi><mo separator="true" lspace="0em" rspace="0em">,</mo></mrow></math>
we observe that <math><mrow><mi>x</mi><mo>&gt;</mo><mn>1</mn><mi>.</mi></mrow></math>
Hence</p>
<p><math display="block" class="tml-display" style="display:block math;"><mrow><mi>x</mi><mo>=</mo><mn>2.62</mn><mspace width="0.2778em"></mspace><mi>■</mi></mrow></math></p>
<p>From <math><mrow><mrow><mi>f</mi><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo><mi>x</mi></mrow><mo separator="true">,</mo></mrow></math>
we can apply <math><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup></math>
on both sides</p>
<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>f</mi><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mi>x</mi></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mi>f</mi><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><msup><mi>f</mi><mrow><mo form="prefix" stretchy="false" lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><msup><mi>f</mi><mrow><mo form="prefix" stretchy="false" lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd></mtr></mtable></math></p>
<p>Hence the value of <math><mi>x</mi></math>
obtained satisfies the equation <math><mrow><mrow><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo><msup><mi>f</mi><mrow><mo form="prefix" stretchy="false" lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false" lspace="0em" rspace="0em">)</mo></mrow><mspace width="0.2778em"></mspace><mi>■</mi></mrow></math></p>
",
          },
        ],
      },
      {
        "parts": [
          {
            "body": "<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mn>2</mn><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>2</mn><mo>+</mo><mi>g</mi><mrow><mo fence="true" form="prefix">(</mo><mn>1</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>2</mn><mo>+</mo><mn>1</mn><mo>+</mo><mi>g</mi><mrow><mo fence="true" form="prefix">(</mo><mn>0</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>3</mn><mo>+</mo><mn>1</mn></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>4</mn></mrow></mtd></mtr></mtable></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mn>4</mn><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>2</mn><mo>+</mo><mi>g</mi><mrow><mo fence="true" form="prefix">(</mo><mn>2</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>2</mn><mo>+</mo><mn>4</mn></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>6</mn><mspace width="0.2778em"></mspace><mi>∎</mi></mrow></mtd></mtr></mtable></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mn>6</mn><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>2</mn><mo>+</mo><mi>g</mi><mrow><mo fence="true" form="prefix">(</mo><mn>3</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>2</mn><mo>+</mo><mn>1</mn><mo>+</mo><mi>g</mi><mrow><mo fence="true" form="prefix">(</mo><mn>2</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>3</mn><mo>+</mo><mn>4</mn></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>7</mn></mrow></mtd></mtr></mtable></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mn>7</mn><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>1</mn><mo>+</mo><mi>g</mi><mrow><mo fence="true" form="prefix">(</mo><mn>6</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>1</mn><mo>+</mo><mn>7</mn></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>8</mn><mspace width="0.2778em"></mspace><mi>∎</mi></mrow></mtd></mtr></mtable></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mn>12</mn><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>2</mn><mo>+</mo><mi>g</mi><mrow><mo fence="true" form="prefix">(</mo><mn>6</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>2</mn><mo>+</mo><mn>2</mn><mo>+</mo><mi>g</mi><mrow><mo fence="true" form="prefix">(</mo><mn>3</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>2</mn><mo>+</mo><mn>7</mn></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>9</mn><mspace width="0.2778em"></mspace><mi>∎</mi></mrow></mtd></mtr></mtable></math></p>
",
          },
          {
            "body": "<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mn>8</mn><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>2</mn><mo>+</mo><mi>g</mi><mrow><mo fence="true" form="prefix">(</mo><mn>4</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>2</mn><mo>+</mo><mn>6</mn></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>8</mn></mrow></mtd></mtr></mtable></math></p>
<p>Since <math><mrow><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mn>7</mn><mo form="postfix" stretchy="false">)</mo><mo>=</mo><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mn>8</mn><mo form="postfix" stretchy="false">)</mo><mo>=</mo><mn>8</mn><mo separator="true" lspace="0em" rspace="0em">,</mo></mrow></math>
<math><mi>g</mi></math>
is not a one-to-one function and <math><mi>g</mi></math>
does not have an inverse<math><mrow><mspace width="0.2778em"></mspace><mi>■</mi></mrow></math></p>
",
          },
        ],
      },
    ],
  },
}
`;
